src/applications/vaos/new-appointment/redux/actions.js

```diff
@@ -262,55 +268,131 @@ export function openFacilityPageV2(page, uiSchema, schema) {
+            parentFacilities.map(parent => {
+              const parentId = parent.id;
+              const siteId = parseFakeFHIRId(
+                getIdOfRootOrganization(parentFacilities, parentId),
+              );
+              return getSupportedHealthcareServicesAndLocations({
+                siteId,
+                parentId,
+                typeOfCareId,
+                useVSP,
+              });
+            }),
           );
-          return getSupportedHealthcareServicesAndLocations({
-            siteId,
-            parentId,
-            typeOfCareId,
-            useVSP,
+          locations = responses
+            .reduce(function(arr, row) {
+              return arr.concat(row.locations);
+            }, [])
+            .sort((a, b) => {
+              return a.name < b.name ? -1 : 1;
+            });
+        } catch (err) {
+          dispatch({
+            type: FORM_FETCH_CHILD_FACILITIES_FAILED,
+            isFacilityV2Page: true,
           });
-        }),
-      );
+        }
+      }
+
+      const eligibilityDataNeeded = !!locationId || locations?.length === 1;
+
+      if (eligibilityDataNeeded && !locationId) {
+        locationId = locations[0].id;
+      }
 
-      const facilities = responses
-        .reduce(function(arr, row) {
-          return arr.concat(row.locations);
-        }, [])
-        .sort((a, b) => {
-          return a.name < b.name ? -1 : 1;
+      if (!locations?.length) {
+        parentFacilities.forEach(p => {
+          recordEligibilityFailure(
+            'supported-facilities',
+            typeOfCare.name,
+            parseFakeFHIRId(p.id),
+          );
         });
+      }
+
+      const eligibilityChecks =
+        newAppointment.eligibility[`${locationId}_${typeOfCareId}`] || null;
+
+      if (eligibilityDataNeeded && !eligibilityChecks) {
+        const parentId = getParentOfLocation(parentFacilities, locations[0]).id;
+        const siteId = parseFakeFHIRId(
+          getIdOfRootOrganization(parentFacilities, parentId),
+        );
+        eligibilityData = await getEligibilityData(
+          locations.find(location => location.id === locationId),
+          typeOfCareId,
+          siteId,
+          directSchedulingEnabled,
+          useVSP,
+        );
+
+        recordEligibilityGAEvents(eligibilityData, typeOfCareId, siteId);
+        logEligibilityExplanation(eligibilityData, typeOfCareId, locationId);
+      }
 
       dispatch({
         type: FORM_PAGE_FACILITY_V2_OPEN_SUCCEEDED,
         page,
         schema,
         uiSchema,
         parentFacilities,
-        facilities,
+        locations,
         typeOfCareId,
+        eligibilityData,
       });
+
+      if (parentFacilities.length && !locations.length) {
+        try {
+          const thunk = fetchFacilityDetails(parentFacilities[0].id);
+          await thunk(dispatch, getState);
+        } catch (e) {
+          captureError(e);
+        }
+      }
```

Comment:
If there are no valid locations, fetch the parent facility info so we can provide the user with contact info for the facility.

**Note:** In the old design, the user will have selected a parent before this contact info is fetched, but since we are fetching all locations, we'll just fetch the info for the first parent. Will create a ticket to request some design guidance on how to update this messaging for the new page.